### PR TITLE
✨ send email when deliberation set the talk to accepted or rejected

### PR DESCRIPTION
**Pull request by corinnekrych**
Related to https://github.com/bpetetot/conference-hall/issues/250
This PR is a proposal to cover the need to send an email for accepted and rejected proposal. The email should be sent only once.

Some topics to discuss:
- maybe we can improve by making the dropdown for the deliberation become readonly once the email was sent.
- is there a way for a speaker to confirm his venue with the application. Fro ex: in the accepted email, I propose a link to the talk. It would be nice to have a confirm button and a new state for the talk "confirmed"

### CODE REVIEW

**Comment by bpetetot** [comment]
@corinnekrych thanks a lot for this PR 🙏 
I'm agree with your proposal, and I will take it next week.
This week I'm working on Devfest Nantes organization, then I will take time to review it.
Thx again ;)

File: functions/email/templates/talkAccepted.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+  <title>✨Your talk has been accepted✨</title>
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0 " />
+</head>
+<body>
+  <p>Dear ${users.map(user => user.displayName)},</p>
```
Comment (frinyvonnick) [inline functions/email/templates/talkAccepted.js]: In this case with two users you will have their names sticked, no ? You could use `join` ? 

Ex: 

```js
users.map(user => user.displayName).join(', ')
```

File: functions/triggers/onUpdateProposal.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    if (!app) return Promise.reject(new Error('You must provide the app.url variable'))
+
+    if (accepted && talk.emailSent === undefined) {
+      console.log(`:::update::accepted & set emailSent to talk ${talk.title}`)
+      // set emailSent to make sure we send accepted email only once.
+      talk.emailSent = talk.updateTimestamp
+      return Promise.all([
+        getEvent(eventId),
+        getUsers(uids),
+        setProposal(eventId, talk)]).then(([event, users]) => sendEmail(mailgun, {
```
Comment (frinyvonnick) [inline functions/triggers/onUpdateProposal.js]: The indentation is hard to read here, no ? Maybe a new line before closing the array would improve it ?

**Review by frinyvonnick — COMMENTED** [review]
👍

**Comment by corinnekrych** [comment]
@frinyvonnick @bpetetot With commit fae02ef I've added a behaviour for the deliberation select box. when orga "accept" or "reject" a talk an email is sent, the select box is greyed to indicate the action can not be undo:
unspecified-11

wdyt?

**Comment by frinyvonnick** [comment]
It looks good to me to reflect the irreversible state of the action, nice add 👍 !

File: src/screens/organizer/components/talkSelection/talkSelection.jsx [hunk 3]
```diff
@@ -1,27 +1,56 @@
...
-    <option key="rejected" value="rejected">
-      Rejected
-    </option>
-  </select>
-)
+const TalkSelection = ({ onChange, defaultValue }) => {
+  const talkDeliberation = defaultValue
+  return (
+    <div>
+      {talkDeliberation === 'accepted' || talkDeliberation === 'rejected' ? (
```
Comment (charlyx) [inline src/screens/organizer/components/talkSelection/talkSelection.jsx:8]: You could save some lines using a variable like `isDisabled` and then use it as a value for `disabled` attribute:

```js
const isDisabled = talkDeliberation === 'accepted' || talkDeliberation === 'rejected'
```

```jsx
<select
  id="ratings"
  onChange={onChange}
  onClick={e => e.stopPropagation()}
  defaultValue={defaultValue}
  disabled={isDisabled}
>
```

**Comment by bpetetot** [comment]
Hi @corinnekrych, you made a great job about this PR, thanks again.
To move foward on it, I propose you to create a feature branch named `feature/deliberation-emails`

We will merge this PR on it then we can work on next steps:
- enable / disable the deliberation email feature in the event configuration page (some organizers prefer to contact speakers themself)
- a way for a speaker to confirm his venue with the application
- update speaker talk status (accepted, rejected) => to be able that the speaker see the new talk status in the app

What do you think about it ?

**Comment by corinnekrych** [comment]
@bpetetot it make sense, I've just rebased this branch from master. if you can create the feature branch, I'll PR this branch (instead of master).
I'll start by the last 2 points as we identifed those too as needed and are required for RivieraDEV cfp ;)
Opt in/out as a configuration sounds good too.

**Comment by bpetetot** [comment]
@corinnekrych I have created the branch:
https://github.com/bpetetot/conference-hall/tree/feature/deliberation-emails
You can base your PR on it.

**Comment by corinnekrych** [comment]
@bpetetot I've created PR: https://github.com/bpetetot/conference-hall/pull/341 to replace this one.

**Pull request closed without merging**